### ISSUE TITLE
va:web:install failure doesn't propagate exit code up to 'yaml-tasks'

### ISSUE THREAD

**Issue by jonpugh** [department-of-veterans-affairs/va.gov-cms#1386]
This is problematic: 

From `tasks.yml`:

```
composer install --dev --no-interaction --no-progress --ansi | tee /dev/stderr | grep 'Could not apply patch' && exit 1 || true`
```

Then it fails down the line.
```
                                                                                                                     
[2/4] Fetching packages...                                                                                           
                                                                                                                     
error https://registry.yarnpkg.com/read-pkg/-/read-pkg-2.0.0.tgz: Extracting tar content of undefined failed, the     ┃ file appears to be corrupt: "ENOENT: no such file or directory, open                                                  ┃ '/var/aegir/.cache/yarn/v4/npm-read-pkg-2.0.0-8ef1c0623c6a6db0dc6713c4bfac46332b2368f8/node_modules/read-pkg/index.js ┃ '"                                                                                                                   
                                                                                                                     
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.                             
                                                                                                                     
Script export NODE_EXTRA_CA_CERTS=/etc/pki/tls/certs/ca-bundle.crt PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=TRUE; cd          ┃ web && yarn install handling the va:web:install event returned with error code 1                                     
                                                                                                                     
Script @va:web:install was called via post-install-cmd                                                               
                                                                                                                     
 ✔ Process Succeeded in 59.507s 


```

**Comment by ElijahLynn** [comment]
We can mock this failure by using https://httptoolkit.tech/, intercepting the response, and changing some bytes in the response. Then a yarn install by itself should fail the same and we can see the error code, and if the error code is the same in a post-install-cmd for composer.

**Comment by oksana-c** [comment]
this may be deprecated or will need to be refined based on outcomes of #3303.

linking to Drupal 9 #1950 epic to keep track of.

**Comment by acrollet** [comment]
This was fixed in #4372
